dbt_style_guide.md

```diff
@@ -1,92 +1,172 @@
+
+- All objects should be plural.  
+  Example: `stg_stripe__invoices.sql` vs. `stg_stripe__invoice.sql`
+
+- All objects should have a prefix to indicate their purpose in the flow:
+  - `base_` is uncommon, but indicates a process needed before `stg_` - typically when multiple sources are rarely used independently. The stg_ model is then used to join or union the sources together before more robust transformations happen.  
+     
+     Example: Creating one table for all cleaned location data.
+     - `base_location__addresses.sql`, `base_location__countries.sql`, and `base_location__states.sql`  would serve to clean the data and be the 1:1 relationship between the source.
+     - `stg_location__locations.sql` would serve to join all location-related data as one entity.
+  - `stg_` is used to clean and standardize data before being fundamentally changed in downstream modeling.
+  - `int_` is used to indicate a step towards creating a final model surfaced to stakeholders
+  - `fct_` is used to indicate a final data set surfaced to stakeholders, and flags data which is in the form of an immutable event stream.
+  - `dim_` is used to indicate a final data set surfaced to stakeholders, and flags data which is used to describe an entity.  
+
+- All models should use the naming convention `<type/dag_stage>_<source/topic>__<additional_context>`. See [this article](https://docs.getdbt.com/blog/stakeholder-friendly-model-names) for more information.
+  - Within the **marts** and **intermediate** folders, `__<additional_context>` is optional. 
+  - Models in the **staging** folder should use the source's name as the `<source/topic>` and the entity name as the `additional_context`.
+
+  Examples:
+  - base_stripe__invoices.sql
+  - stg_stripe__customers.sql
+  - int_payments.sql
+  - int_customers__unioned.sql
+  - fct_orders.sql
+
+### Organization
+Our models (typically) fit into three main categories: staging, marts, and base/intermediate.  
+
+For more detail about why we use this structure, check out [this discourse post](https://discourse.getdbt.com/t/how-we-structure-our-dbt-projects/355). The file and naming structures are as follows:
 ```
 ├── dbt_project.yml
 └── models
     ├── marts
-    |   └── core
-    |       ├── intermediate
-    |       |   ├── intermediate.yml
-    |       |   ├── customers__unioned.sql
-    |       |   ├── customers__grouped.sql
-    |       └── core.yml
-    |       └── core.docs
-    |       └── dim_customers.sql
-    |       └── fct_orders.sql
+    |   ├── core
+    |   |   ├── _docs_core.md
+    |   |   ├── _models_core.yml
+    |   |   ├── dim_customers.sql
+    |   |   └── fct_orders.sql
+    |   └── intermediate
+    |       ├── _models_int_core.yml
+    |       ├── int_payments.sql
+    |       ├── int_customers__unioned.sql
+    |       └── int_customers__grouped.sql

```

Comment:
Everything LGTM! The only comment I have and I don't know how other people feel about this: I am not sure I agree with the base_ and int_ YAML files. I feel like subdirectories are often small enough that a single model_ YAML file can exist for that and its int_ or base_ models. I would recommend it only to very large projects. Interested to hear other peoples' thoughts.

Reply:
Voting: I'm leaning more towards recommending this vs. not recommending it and keeping it at the parent level. Projects only grow over time, so this is a good setup for future-proofing. However, I really have no strong preference on this. We'll need to change what it says under the yml section further below but keep some examples here as a visual to how these should be named if one decides to use them within further subdirectories.

Reply:
No strong preference on this. But to cast my vote, I like the idea of future-proofing, so leaving them as separate files feels fine to me.

Reply:
I like to have 1 `yml` model file per folder, even if there are only a few models. I find it easier to find where the `yml` file of a specific model is.

Reply:
Keeping this as is based on comments.